### ISSUE TITLE
Popup menus and tooltips are not working when no displays connected

### ISSUE THREAD

**Issue by andreymal** [godotengine/godot#85886]
### Tested versions

4.2.stable

### System information

Linux, X11

### Issue description

I'm connecting to my PC remotely via VNC, and since the physical display is off, the system thinks there are no displays connected. Godot complains about it:

```
platform/linuxbsd/x11/display_server_x11.cpp:1062 - Invalid screen index: 0(count: 0).
platform/linuxbsd/x11/display_server_x11.cpp:1133 - Index p_screen = 0 is out of bounds (screen_count = 0).
```

Most functions work fine, however I have noticed that pop-up menus and tooltips don't work. I suspect that the menus are 0x0 pixels in size and therefore not visible.

However, the X11 screen still exists even without display, and I think Godot should work with it correctly.

![](https://github.com/godotengine/godot/assets/3236464/269c28d2-b3e7-42f4-a249-b918d3a05696)

### Steps to reproduce

Unplug the display cable, I guess?

### Minimal reproduction project (MRP)

N/A

**Comment by AlbertoMQ** [comment]
I still have this same issue when connecting to a remote machine using TeamViewer

**Comment by AThousandShips** [comment]
You are using 4.3.beta1?

**Comment by AlbertoMQ** [comment]
Ope, thought I was using the newest release, but I wasn't. Issue is fixed in 4.3
